src/components/Header.tsx

```diff
@@ -0,0 +1,41 @@
+import { useColorMode, IconButton, Container } from '@chakra-ui/react';
+import { SunIcon, MoonIcon } from '@chakra-ui/icons';
+
+export function Header() {
+  const { colorMode, toggleColorMode } = useColorMode();
+
+  return (
+    <Container
+      maxW="160ch"
+      justifyContent={'end'}
+      alignItems={'end'}
+      display={'flex'}
+    >
+      {/* IconButton */}
+      {colorMode === 'light' ? (
+        <IconButton
+          aria-label="Switcher"
+          justifyItems={'end'}
+          my="20px"
+          m="8"
+          isRound
+          color="gray.800"
+          icon={<MoonIcon />}
+          variant="outline"
+          onClick={toggleColorMode}
+        />
+      ) : (
+        <IconButton
+          aria-label="Switcher"
+          my="20px"
+          m="8"
+          isRound
+          color="yellow.500"
+          icon={<SunIcon />}
+          variant="outline"
+          onClick={toggleColorMode}
+        />
+      )}
+    </Container>
+  );
+}
```

Comment:
```suggestion
import { useColorMode, IconButton, Container, useColorModeValue } from '@chakra-ui/react';
import { SunIcon, MoonIcon } from '@chakra-ui/icons';

export function Header() {
  const { toggleColorMode } = useColorMode();
  const { color, icon } = useColorModeValue({ color: 'gray.800', icon: <MoonIcon /> }, { color: 'yellow.500', icon: <SunIcon /> })

  return (
    <Container
      maxW="160ch"
      justifyContent="end"
      alignItems="end"
      display="flex"
    >
      <IconButton
        aria-label="Switcher"
        m="8"
        isRound
        color={color}
        icon={icon}
        variant="outline"
        onClick={toggleColorMode}
      />
    </Container>
  );
}
```

Would be cool to use the useColorModeValue to manage the color and icon instead of an IF, it prevents code duplication.
I think that is better to have some standard when passing props, like curly braces only if you are using some ResponsiveValue or variables and keep the double quotes for strings.
`<Button size="lg" color={{ base: 'red', md: 'blue' }} />`

One more thing the my prop the way you did is being overridden by the m props, the order matters here so if you like to make the my you should change to my come after m.